File: src/code/java/Q136.java
package code.java;

public class Q136 {
    // 原理: a^0=a, a^a=0, a^b^a=a^a^b=b
    public int singleNumber(int[] nums) {
        int n = 0;
        for (int num : nums) {
            n ^= num;
        }
        return n;
    }
}

File: src/code/js/Q136.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function (nums) {
    //原理: a^0=a, a^a=0, a^b^a=a^a^b=b
    var n = 0;
    for (var i = 0; i < nums.length; i++) {
        n ^= nums[i];
    }
    return n;
};